src/Sarif.Converters/PREFastConverter.cs

```diff
@@ -0,0 +1,193 @@
+            var defectList = (DefectList)serializer.Deserialize(input);
+            var results = new List<Result>();
+            foreach (Defect entry in defectList.Defects)
+            {
+                results.Add(CreateResult(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        private Result CreateResult(Defect defect)
+        {
+            var region = new Region
+            {
+                StartColumn = defect.SFA.Column + 1,
+                StartLine = defect.SFA.Line
+            };
+
+            Uri resultsFileUri = new Uri($"{defect.SFA.FilePath}{defect.SFA.FileName}", UriKind.Relative);
+            PhysicalLocation physicalLocation = new PhysicalLocation(uri: resultsFileUri, uriBaseId: null, region: region);
+            Location location = new Location()
+            {
+                ResultFile = physicalLocation,
+                FullyQualifiedLogicalName = defect.Function,
+                DecoratedName = defect.Decorated
+            };
+
+            location.SetProperty("funcline", defect.Funcline);
+
+            var result = new Result
+            {
+                RuleId = defect.DefectCode,
+                Message = RemoveLegacyNewLine(defect.Description),
+                Locations = new List<Location>()
+            };
+
+            result.Locations.Add(location);
+            SetProbability(defect, result);
+            SetRank(defect, result);
+
+            ExtractCategories(defect, result);
+            GenerateCodeFlows(defect, result);
+
+            return result;
+        }
+
+        private void SetRank(Defect defect, Result result)
+        {
+            if (string.IsNullOrWhiteSpace(defect.Rank))
+                return;
+
+            result.SetProperty("rank", defect.Rank);
+        }
+
+        private void SetProbability(Defect defect, Result result)
+        {
+            if (string.IsNullOrWhiteSpace(defect.Probability))
+                return;
+
+            result.SetProperty("probability", defect.Probability);
+        }
+
+        private void GenerateCodeFlows(Defect defect, Result result)
+        {
+            if (defect.Path?.SFAs.Any() == false)
+                return;
+
+            int step = 0;
+            var locations = new List<AnnotatedCodeLocation>();
+            bool pathUsesKeyEvents = defect.Path.SFAs.Any(x => !string.IsNullOrWhiteSpace(x?.KeyEvent?.Id));
+
+            foreach (var sfa in defect.Path.SFAs)
+            {
+                var reg = new Region()
```

Comment:
`reg` => `region`

Reply:
fixed

---
In reply to: [146951857](https://github.com/Microsoft/sarif-sdk/pull/657#discussion_r146951857) [](ancestors = 146951857)